src/main/java/org/zalando/nakadi/service/EventPublisher.java

```diff
@@ -215,17 +207,21 @@ private void validate(final List<BatchItem> batch, final EventType eventType, fi
             throws EventValidationException, InternalNakadiException, NoSuchEventTypeException {
         final Span validationSpan = TracingService.getNewSpanWithParent("validation", System.currentTimeMillis(),
                 parentSpan);
-        TracingService.setCustomTags(validationSpan, ImmutableMap.<String, Object>builder()
-                .put("event_type", eventType).build());
-        try (Scope validationScope = TracingService.activateSpan(validationSpan, false)) {
+        validationSpan.setTag("event_type", eventType.getName());
+        try {
             for (final BatchItem item : batch) {
                 item.setStep(EventPublishingStep.VALIDATING);
                 try {
                     validateSchema(item.getEvent(), eventType);
                     validateEventSize(item);
                 } catch (final EventValidationException e) {
                     item.updateStatusAndDetail(EventPublishingStatus.FAILED, e.getMessage());
-                    traceEventPublishingError(validationScope, item, e.getMessage());
+                    if (eventType.getCategory() != EventCategory.UNDEFINED) {
+                        validationSpan.log(ImmutableMap.of(
```

Comment:
I think this is another time bomb. If we get a lot of failures, then all of a sudden we allocate a lot more objects in order to log something that is not even that important (event ids). This information is also in the response body. @adyach I would remove from live 219 until 224.

Reply:
Ok, I see that this is a very common question from users. Let's try with it for some time.